lib/actions/status.js

```diff
@@ -2,28 +2,50 @@ const { find } = require("lodash");
 const migrationsDir = require("../env/migrationsDir");
 const config = require("../env/config");
 
-module.exports = async db => {
+module.exports = async (db, shouldLoadFromDatabase) => {
   await migrationsDir.shouldExist();
   await config.shouldExist();
   const fileNames = await migrationsDir.getFileNames();
 
-  const { changelogCollectionName, useFileHash } = await config.read();
+  const { changelogCollectionName, useFileHash, saveFileContents } = await config.read();
   const changelogCollection = db.collection(changelogCollectionName);
   const changelog = await changelogCollection.find({}).toArray();
 
-
   const useFileHashTest = useFileHash === true;
   const statusTable = await Promise.all(fileNames.map(async (fileName) => {
     let fileHash;
-    let findTest = { fileName };
+    const findTest = { fileName };
     if (useFileHashTest) {
       fileHash = await migrationsDir.loadFileHash(fileName);
-      findTest = { fileName, fileHash };
+      findTest.fileHash = fileHash;
     }
     const itemInLog = find(changelog, findTest);
-    const appliedAt = itemInLog ? itemInLog.appliedAt.toJSON() : "PENDING";
-    return useFileHash ? { fileName, fileHash, appliedAt } : { fileName, appliedAt };
+    if (itemInLog) {
+      const indexInLog = changelog.indexOf(itemInLog);
+      changelog.splice(indexInLog, 1);
+    }
+    findTest.appliedAt = itemInLog ? itemInLog.appliedAt.toJSON() : "PENDING";
+    if (saveFileContents) {
+      const fileContents = await migrationsDir.loadFileContents(fileName);
+      findTest.fileContents = fileContents.toString();
+    }
+    return findTest;
   }));
 
+  // if we have saved migrations in the database, we need to add any migration which is in the database without a file
+  if (shouldLoadFromDatabase === true) {
+    changelog.forEach(migrationInDb => {
```

Comment:
Assuming that changelog is an array, I think a for of loop is more declarative
```suggestion
    for (const migrationInDb of changelog) {
```

Reply:
I tried changing this but ESLint complained saying that I should not use for loops in this case!